stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/components/avatar/Avatar.kt

```diff
@@ -1,50 +1,128 @@
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import coil.compose.ImagePainter
+import coil.compose.rememberImagePainter
+import io.getstream.chat.android.compose.R
 import io.getstream.chat.android.compose.ui.theme.ChatTheme
 
 /**
- * Avatar/image that renders whatever image the [painter] provides. It allows for customization,
- * uses the 'avatar' shape from [ChatTheme.shapes] for the clipping and exposes an [onClick] action.
+ * An avatar that renders an image from the provided image URL. In case the image URL
+ * was empty or there was an error loading the image, it falls back to the initials avatar.
  *
- * @param painter The painter for the image.
+ * @param imageUrl The URL of the image to load.
+ * @param initials The fallback text.
  * @param modifier Modifier for styling.
  * @param shape The shape of the avatar.
+ * @param textStyle The text style of the [initials] text.
  * @param contentDescription Description of the image.
  * @param onClick OnClick action, that can be nullable.
  */
 @Composable
 public fun Avatar(
-    painter: Painter,
+    imageUrl: String,
+    initials: String,
     modifier: Modifier = Modifier,
     shape: Shape = ChatTheme.shapes.avatar,
+    textStyle: TextStyle = ChatTheme.typography.title3Bold,
     contentDescription: String? = null,
     onClick: (() -> Unit)? = null,
 ) {
-    val clickableModifier: Modifier = if (onClick != null) {
-        modifier.clickable(
-            onClick = onClick,
-            indication = rememberRipple(bounded = false),
-            interactionSource = remember { MutableInteractionSource() }
+    if (LocalInspectionMode.current && imageUrl.isNotBlank()) {
+        // Show hardcoded avatar from resources when rendering previews
+        ImageAvatar(
+            modifier = modifier,
+            shape = shape,
+            painter = painterResource(id = R.drawable.stream_compose_preview_avatar),
+            contentDescription = contentDescription,
+            onClick = onClick
+        )
+        return
+    }
+    if (imageUrl.isNullOrBlank()) {
+        InitialsAvatar(
+            modifier = modifier,
+            initials = initials,
+            shape = shape,
+            textStyle = textStyle,
+            onClick = onClick
+        )
+        return
+    }
+
+    val painter = rememberImagePainter(data = imageUrl)
+
+    if (painter.state is ImagePainter.State.Error) {
+        InitialsAvatar(
+            modifier = modifier,
+            initials = initials,
+            shape = shape,
+            textStyle = textStyle,
+            onClick = onClick
         )
     } else {
-        modifier
+        ImageAvatar(
+            modifier = modifier,
+            shape = shape,
+            painter = painter,
+            contentDescription = contentDescription,
+            onClick = onClick
+        )
```

Comment:
Just small a suggestion from an idea. 😄  

How about handling each case of the `ImagePainter.State` like the below?
Maybe we can consider receiving a `placeHolderPainter` parameter which is the type of the `Painter`, so clients can show a placeholder image when loading the images. (I guess if we're in a poor network environment, it could be taken a few seconds for loading an image.)

```kotlin
when (painter.state) {
        is ImagePainter.State.Error -> {
            InitialsAvatar(
                modifier = modifier,
                initials = initials,
                shape = shape,
                textStyle = textStyle,
                onClick = onClick
            )
        }
        is ImagePainter.State.Success, ImagePainter.State.Empty -> {
            ImageAvatar(
                modifier = modifier,
                shape = shape,
                painter = painter,
                contentDescription = contentDescription,
                onClick = onClick
            )
        }
        is ImagePainter.State.Loading -> {
            ImageAvatar(
                modifier = modifier,
                shape = shape,
                painter = placeHolderPainter,
                contentDescription = contentDescription,
                onClick = onClick
            )
        }
    }
```
Also, we may consider making it smooth by wrapping the above code with `Crossfade(targetState = painter.state)`.

Reply:
This is a great suggestion - if we're using this painter.state API, then we can do this and we can add the placeholderImage param that's empty or defined by us, by default.

Reply:
Done, thanks for the suggestion!